api/src/main/java/org/eclipse/microprofile/rest/client/RestClientBuilder.java

```diff
@@ -50,11 +51,36 @@ static RestClientBuilder newBuilder() {
      * <code>http://my-service:8080/service/api</code> in addition to any
      * <code>@Path</code> annotations included on the method.
      *
+     * The {@link #baseUri(URI)} method should be preferred over this method.
+     *
+     * Subsequent calls to this method will replace the previously specified
+     * baseUri/baseUrl.
+     *
      * @param url the base Url for the service.
      * @return the current builder with the baseUrl set
+     * @deprecated use {@link #baseUri(java.net.URI)} instead.
      */
+    @Deprecated
     RestClientBuilder baseUrl(URL url);
 
+    /**
+     * Specifies the base URI to be used when making requests. Assuming that the
+     * interface has a <code>@Path("/api")</code> at the interface level and a
+     * <code>uri</code> is given with
+     * <code>http://my-service:8080/service</code> then all REST calls will be
+     * invoked with a <code>uri</code> of
+     * <code>http://my-service:8080/service/api</code> in addition to any
+     * <code>@Path</code> annotations included on the method.
+     *
+     * Subsequent calls to this method will replace the previously specified
+     * baseUri/baseUrl.
+     *
+     * @param uri the base URI for the service.
+     * @return the current builder with the baseUri set
+     * @since 1.1
+     */
+    RestClientBuilder baseUri(URI uri);
```

Comment:
To avoid API breakage, can this be a default method that performs `return baseUrl(uri.toURL())`?

Reply:
Yes, great suggestion!